libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -242,9 +259,21 @@ bool AP_Camera::update_location(const struct Location &loc)
         // be called without a new GPS fix
         return false;
     }
+
     if (get_distance(loc, _last_location) < _trigg_dist) {
         return false;
     }
-    _last_location = loc;
-    return true;
+
+    if ((unsigned) ahrs.roll_sensor/100.0f > _max_roll) {
```

Comment:
Oh yeah, totally forgot about that when I wrote the above comment... Yeah if possible sticking with int math is probably better.